features/vaadin-dashboard/src/main/java/org/opennms/features/vaadin/dashboard/config/ui/editors/CriteriaBuilderHelper.java

```diff
@@ -128,15 +132,44 @@ public OnmsSeverity getDefault() {
         });
 
         setCriteriaParser(Date.class, new CriteriaParser<Date>() {
+            final DateTimeFormatter parserFormatter = new DateTimeFormatterBuilder()
+                    .append(DateTimeFormatter.ISO_OFFSET_DATE_TIME)
+                    .parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+                    .parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+                    .toFormatter();
+
+            final String DELTA_PATTERN = "^[+-]\\d+";
+
+            private ZonedDateTime parseZonedDateTinme(final String string) {
+                if ("0".equals(string)) {
+                    return ZonedDateTime.now();
+                } else {
+                    if (string.matches(DELTA_PATTERN)) {
+                        final long delta = Long.parseLong(string.substring(1));
+                        if (string.charAt(0) == '+') {
+                            return ZonedDateTime.now().plus(delta, ChronoUnit.SECONDS);
+                        } else {
+                            return ZonedDateTime.now().minus(delta, ChronoUnit.SECONDS);
+                        }
+                    } else {
+                        try {
+                            return ZonedDateTime.parse(string, parserFormatter);
+                        } catch (DateTimeParseException ex) {
+                            return null;
```

Comment:
Do we log at some point, that the provided expression could not be parsed? Maybe helpful for users.

Reply:
Yes, it will be logged outside.

Reply:
Perfect 👍